crates/ruff/src/rules/flake8_implicit_str_concat/rules/explicit.rs

```diff
@@ -38,15 +39,15 @@ impl Violation for ExplicitStringConcatenation {
 }
 
 /// ISC003
-pub(crate) fn explicit(expr: &Expr) -> Option<Diagnostic> {
+pub(crate) fn explicit(expr: &Expr, locator: &Locator) -> Option<Diagnostic> {
     if let Expr::BinOp(ast::ExprBinOp {
         left,
         op,
         right,
-        range: _,
+        range,
     }) = expr
     {
-        if matches!(op, Operator::Add) {
+        if locator.contains_line_break(*range) && matches!(op, Operator::Add) {
```

Comment:
Nit: Can we make this the last check (move it after the `left` and `right` checks as well). Testing whether the range contains a line break isn't extremely expensive, but is more expensive than testing whether the operator or `left` and `right` are of a certain type.